File: 数据结构与算法/极客时间/链表/实现/LinkedList.py
class Node:
    def __init__(self, val = None, next = None):
        self.val = val
        self.next = next

class ListNode:
    def __init__(self):
        self.head = Node();

    def insertAfter(self,sourceNode, targetNode = None):
        if not self.head.val and not self.head.next:
            self.head = sourceNode
        else:
            sourceNode.next = targetNode.next
            targetNode.next = sourceNode
        return self.head

    def removeNode(self, node):
        first = self.head
        while first.next:
            if first.next is node:
                first.next = first.next.next
                break
            first = first.next
        return self.head

a = Node(1)
b = Node(2)
c = Node(3)
list = ListNode()
list.insertAfter(a)
list.insertAfter(b, a)
list.insertAfter(c, b)
list.removeNode(b)
print(list)
print(__name__)

File: 数据结构与算法/极客时间/链表/实现/LinkedList.js
function Node(data = null, next = null) {
  this.data = data;
  this.next = next;
}

function ListNode() {
  this.head = new Node();
}
ListNode.prototype.insertAfter = function (sourceNode, targetNode) {
  let first = this.head;
  if (first.data === null && first.next === null) {
    this.head = sourceNode;
  } else {
    sourceNode.next = targetNode.next;
    targetNode.next = sourceNode;
  }
  return first;
}
ListNode.prototype.removeNode = function (node) {
  let first = this.head;
  /** 找到 node 节点的父节点 */
  while (first.next) {
    if (first.next === node) {
      first.next = first.next.next;
      break;
    }
    first = first.next;
  }
  return first;
}

var a = new Node(1)
var b = new Node(2)
var c = new Node(3)
var list = new ListNode(a)
list.insertAfter(a)
list.insertAfter(b, a)
list.insertAfter(c, b)
list.removeNode(b)
console.log(list)
